### PR TITLE
Improvization of carousel via removing extra sliding line in js plus …

### LINKED ISSUE #61: Carousel and Hero H1 H2 title positioning
Screenshot 2023-07-05 at 14 34 48
Screenshot 2023-07-06 at 07 15 47

### PR DESCRIPTION
**Pull request by meejain**
…no need of absolute position for buttons

Fix #61 

Test URLs:
- Before: https://main--ororafresh--hlxsites.hlx.page/
- After: https://hero-carousel-17--ororafresh--hlxsites.hlx.page/Drafts/Meet/

### CODE REVIEW

**Comment by meejain** [comment]
Hello @paolomoz , 

Seems like these 2 lines are not required - 

block.scrollTo({ top: 0, left: row.offsetLeft - row.parentNode.offsetLeft, behavior: 'smooth' }); in carousel.js

& position: absolute; for main .carousel-wrapper .carousel-buttons  in. carousel.css

1 query here - Could notice, div button position if absolute - ideally should check its closest parent to be relative. There is no parent div which has got the position relative & hence ideally it should position itself as per window. Still its positioning itself below the carousel block div. Need an understanding here . Please advise. 

image

**Comment by meejain** [comment]
@paolomoz  - Changed the position & responsiveness of H1 & H2 text within the carousel. Its a lot better now. Please check & validate - 

https://hero-carousel-17--ororafresh--hlxsites.hlx.page/Drafts/Meet/

Many Thanks

**Comment by meejain** [comment]
@paolomoz  - Noticed that https://ororafresh.com/ - There carousel on Hero is automatically switching the images after 10 seconds, hence implemented a logic. 

Please check & validate - https://hero-carousel-17--ororafresh--hlxsites.hlx.page/Drafts/Meet/

File: blocks/carousel/carousel.js [hunk 1]
```diff
@@ -20,4 +20,8 @@ export default function decorate(block) {
     buttons.append(button);
   });
   block.parentElement.append(buttons);
+  const button0 = document.getElementById('carousel-button-0');
```
Comment (paolomoz) [inline blocks/carousel/carousel.js:23]: this should be recursive regardless the number of items in the carousel. I suggest to test how the whole carousel works with 1,2,3 or 4 items if everything works as expected

**Review by paolomoz — CHANGES_REQUESTED** [review]

File: blocks/carousel/carousel.css [hunk 2]
```diff
@@ -66,7 +66,8 @@ main .carousel-wrapper {
 main .carousel-wrapper .carousel-buttons {
     text-align: center;
     margin-top: -40px;
-    position: absolute;
+
+    /* position: absolute; */
```
[the remark this answers is not in the document]
Comment (meejain) [reply blocks/carousel/carousel.css:70]: Fixed

File: blocks/carousel/carousel.js (hunk 1 above)
Comment (meejain) [reply blocks/carousel/carousel.js:23]: Fixed.

File: blocks/carousel/carousel.css [hunk 3]
```diff
@@ -213,4 +213,13 @@ main .carousel-image picture:last-of-type img {
     main .carousel-image picture:first-of-type img {
         display: block;
     }
+    
+}
+
+
+@media (min-width: 1800px) {
```
[the remark this answers is not in the document]
Comment (meejain) [reply blocks/carousel/carousel.css]: I did test that on an imac > 1500px where the 2nd image is looking like this, hence I added that. For now I removed that piece of code from css. 

image

[the remark this answers is not in the document]
Comment (paolomoz) [reply blocks/carousel/carousel.css]: ok yes, I could reproduce it. the real problem is that font size is dynamic on vw and that makes fonts very big on very big screens. For body text we set a fixed font size; we should set a fixed font size for everything (including titles and hero and carousel for (min-width: 1200px) so the H1 and H2 of the carousel don't become bigger than that size and you don't nered a bigger line-height, hence text dowsn't overlay between lines

[the remark this answers is not in the document]
Comment (meejain) [reply blocks/carousel/carousel.css]: @paolomoz  . Its fixed now with the firm values. Tried and tested. Looks better. Ty again.

**Comment by meejain** [comment]
"On large screen the H1 and H2 titles are aligned on the left while they should stay within the content width, see picture below."

@paolomoz  - Done with the changes via making margin of the containing DIV as auto and width as 50%. Please check & validate further - 

https://hero-carousel-17--ororafresh--hlxsites.hlx.page/Drafts/Meet/

Thank You, 

Meet

**Review by paolomoz — APPROVED** [review]

**Pull request merged**